src/app/lib/utilities/preprocessor/rules/addIdsToBlocks.js

```diff
@@ -38,13 +38,12 @@ const mergeJsonRawWithBlocks = blocksWithIds => jsonRaw => ({
   },
 });
 
-export default jsonRaw => {
-  const addIdsToBlocks = compose(
+const addIdsToBlocks = async jsonRaw =>
+  compose(
     mergeJsonRawWithBlocks,
     mapIdsToBlocks,
     getBlocks,
     getJsonContent,
-  )(jsonRaw);
+  )(await jsonRaw)(await jsonRaw);
```

Comment:
Not 100% sure why this is needed

Reply:
This is gross, not even much better like:
```
const addIdsToBlocks = async jsonRaw => {
  const data = await jsonRaw;
  return compose(
    mergeJsonRawWithBlocks,
    mapIdsToBlocks,
    getBlocks,
    getJsonContent,
  )(data)(data);
};
```

Reply:
The `(data)(data)` is soooo strange. @jroebu14 any ideas? :P

Reply:
@jroebu14 any thoughts on this `compose`?